megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -1415,8 +1415,7 @@ protected boolean isSubCommander(final Entity entity) {
         } else if (0.5 > damageFraction
                    || Targetable.TYPE_BUILDING == target.getTargetType()
                    || Targetable.TYPE_HEX_CLEAR == target.getTargetType()
-                   || owner.getGame().getEntity(target.getTargetId()) instanceof Infantry
-                   || owner.getGame().getEntity(target.getTargetId()) instanceof BattleArmor) {
+                   || owner.getGame().getEntity(target.getTargetId()) instanceof Infantry) {
```

Comment:
The BattleArmor check here was checking if it was BattleArmor after infantry, which would have already been caught if true by the infantry check.